testing/test-nx-utils/src/lib/utils/nx.unit.test.ts

```diff
@@ -0,0 +1,84 @@
+import * as process from 'node:process';
+import { createTreeWithEmptyWorkspace } from 'nx/src/generators/testing-utils/create-tree-with-empty-workspace';
+import { describe, expect } from 'vitest';
+import { executorContext, registerPluginInWorkspace } from './nx';
+
+describe('executorContext', () => {
+  it('should create context for given project name', () => {
+    expect(executorContext('my-lib')).toStrictEqual({
```

Comment:
⚙️ TypeScript: To get auto-complete if there is a non-trivial type to match, you can use `satisfies` or a type argument. Applicable in multiple places.